The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14685096065/job/41212709304) [(plain)](https://github.com/rust-lang/rust/commit/80ca3bf528b7248b1f235e632232262ea2c91c3f/checks/41212709304/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04

ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
#
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 2.766 Building wheels for collected packages: reuse
#12 2.767   Building wheel for reuse (pyproject.toml): started
#12 2.981   Building wheel for reuse (pyproject.toml): finished with status 'done'
#12 2.982   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132719 sha256=5bb60f62728aaedff7162745ce743c7f2f55069b3e7f82e6a37d70df455797cc
#12 2.983   Stored in directory: /tmp/pip-ephem-wheel-cache-fhc7_qjt/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#12 2.985 Successfully built reuse
#12 2.986 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#12 3.392 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#12 3.392 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 3.927 Collecting virtualenv
#12 3.968   Downloading virtualenv-20.30.0-py3-none-any.whl (4.3 MB)
#12 4.035      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 67.1 MB/s eta 0:00:00
#12 4.075 Collecting distlib<1,>=0.3.7
#12 4.079   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.086      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 96.0 MB/s eta 0:00:00
#12 4.120 Collecting platformdirs<5,>=3.9.1
#12 4.124   Downloading platformdirs-4.3.7-py3-none-any.whl (18 kB)
#12 4.161 Collecting filelock<4,>=3.12.2
#12 4.166   Downloading filelock-3.18.0-py3-none-any.whl (16 kB)
#12 4.250 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.436 Successfully installed distlib-0.3.9 filelock-3.18.0 platformdirs-4.3.7 virtualenv-20.30.0
#12 4.436 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 DONE 4.5s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
DirectMap4k:      120768 kB
DirectMap2M:     7219200 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
[TIMING] core::build_steps::tool::LibcxxVersionTool { target: x86_64-unknown-linux-gnu } -- 0.222
---
fmt check
fmt: checked 5998 files
tidy check
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/src/ci/scripts/free-disk-space.sh:149: line longer than 100 chars
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'venv'
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'virtualenv'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (25.0.1)
Collecting pip
  Downloading pip-25.1-py3-none-any.whl.metadata (3.6 kB)
Downloading pip-25.1-py3-none-any.whl (1.8 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 MB 106.8 MB/s eta 0:00:00
Installing collected packages: pip
---
All checks passed!
checking python file formatting
26 files already formatted
checking C++ file formatting
some tidy checks failed
Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:01:59
  local time: Sat Apr 26 21:15:55 UTC 2025
  network time: Sat, 26 Apr 2025 21:15:55 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/ci/github-actions/jobs.yml
+++ b/src/ci/github-actions/jobs.yml
@@ -65,6 +65,12 @@ runners:
     os: codebuild-ubuntu-22-36c-$github.run_id-$github.run_attempt
     <<: *base-job
 
+  - &job-linux-8c-codebuild
+    free_disk: true
+    codebuild: true
+    os: codebuild-ubuntu-22-8c-$github.run_id-$github.run_attempt
+    <<: *base-job
+
 envs:
   env-x86_64-apple-tests: &env-x86_64-apple-tests
     SCRIPT: ./x.py check compiletest --set build.compiletest-use-stage0-libtest=true && ./x.py --stage 2 test --skip tests/ui --skip tests/rustdoc -- --exact
@@ -160,7 +166,7 @@ auto:
     <<: *job-linux-4c
 
   - name: dist-arm-linux
-    <<: *job-linux-36c-codebuild
+    <<: *job-linux-8c-codebuild
 
   - name: dist-armhf-linux
     <<: *job-linux-4c

--- a/src/ci/scripts/free-disk-space.sh
+++ b/src/ci/scripts/free-disk-space.sh
@@ -87,59 +87,80 @@ printDF() {
 
 removeUnusedFilesAndDirs() {
     local to_remove=(
-        "/usr/local/aws-sam-cli"
-        "/usr/local/doc/cmake"
-        "/usr/local/julia"*
-        "/usr/local/lib/android"
-        "/usr/local/share/chromedriver-"*
-        "/usr/local/share/chromium"
-        "/usr/local/share/cmake-"*
-        "/usr/local/share/edge_driver"
-        "/usr/local/share/gecko_driver"
-        "/usr/local/share/icons"
-        "/usr/local/share/vim"
-        "/usr/local/share/emacs"
-        "/usr/local/share/powershell"
-        "/usr/local/share/vcpkg"
-        "/usr/share/apache-maven-"*
-        "/usr/share/gradle-"*
         "/usr/share/java"
-        "/usr/share/kotlinc"
-        "/usr/share/miniconda"
-        "/usr/share/php"
-        "/usr/share/ri"
-        "/usr/share/swift"
-
-        # binaries
-        "/usr/local/bin/azcopy"
-        "/usr/local/bin/bicep"
-        "/usr/local/bin/ccmake"
-        "/usr/local/bin/cmake-"*
-        "/usr/local/bin/cmake"
-        "/usr/local/bin/cpack"
-        "/usr/local/bin/ctest"
-        "/usr/local/bin/helm"
-        "/usr/local/bin/kind"
-        "/usr/local/bin/kustomize"
-        "/usr/local/bin/minikube"
-        "/usr/local/bin/packer"
-        "/usr/local/bin/phpunit"
-        "/usr/local/bin/pulumi-"*
-        "/usr/local/bin/pulumi"
-        "/usr/local/bin/stack"
-
-        # Haskell runtime
-        "/usr/local/.ghcup"
-
-        # Azure
-        "/opt/az"
-        "/usr/share/az_"*
     )
 
-    if [ -n "${AGENT_TOOLSDIRECTORY:-}" ]; then
-        # Environment variable set by GitHub Actions
+    if isGitHubRunner; then
         to_remove+=(
-            "${AGENT_TOOLSDIRECTORY}"
+            "/usr/local/aws-sam-cli"
+            "/usr/local/doc/cmake"
+            "/usr/local/julia"*
+            "/usr/local/lib/android"
+            "/usr/local/share/chromedriver-"*
+            "/usr/local/share/chromium"
+            "/usr/local/share/cmake-"*
+            "/usr/local/share/edge_driver"
+            "/usr/local/share/emacs"
+            "/usr/local/share/gecko_driver"
+            "/usr/local/share/icons"
+            "/usr/local/share/powershell"
+            "/usr/local/share/vcpkg"
+            "/usr/local/share/vim"
+            "/usr/share/apache-maven-"*
+            "/usr/share/gradle-"*
+            "/usr/share/kotlinc"
+            "/usr/share/miniconda"
+            "/usr/share/php"
+            "/usr/share/ri"
+            "/usr/share/swift"
+
+            # binaries
+            "/usr/local/bin/azcopy"
+            "/usr/local/bin/bicep"
+            "/usr/local/bin/ccmake"
+            "/usr/local/bin/cmake-"*
+            "/usr/local/bin/cmake"
+            "/usr/local/bin/cpack"
+            "/usr/local/bin/ctest"
+            "/usr/local/bin/helm"
+            "/usr/local/bin/kind"
+            "/usr/local/bin/kustomize"
+            "/usr/local/bin/minikube"
+            "/usr/local/bin/packer"
+            "/usr/local/bin/phpunit"
+            "/usr/local/bin/pulumi-"*
+            "/usr/local/bin/pulumi"
+            "/usr/local/bin/stack"
+
+            # Haskell runtime
+            "/usr/local/.ghcup"
+
+            # Azure
+            "/opt/az"
+            "/usr/share/az_"*
+        )
+
+        if [ -n "${AGENT_TOOLSDIRECTORY:-}" ]; then
+            # Environment variable set by GitHub Actions
+            to_remove+=(
+                "${AGENT_TOOLSDIRECTORY}"
+            )
+        else
+            echo "::warning::AGENT_TOOLSDIRECTORY is not set. Skipping removal."
+        fi
+    else
+        # Remove folders and files present in AWS CodeBuild
+        to_remove+=(
+            # binaries
+            "/usr/local/bin/ecs-cli"
+            "/usr/local/bin/eksctl"
+            "/usr/local/bin/kubectl"
+
+            "${HOME}/.gradle"
+            "${HOME}/.dotnet"
+            "${HOME}/.goenv"
+            "${HOME}/.phpenv"
+
         )
     fi
 
@@ -194,6 +215,10 @@ cleanPackages() {
                 'powershell'
             )
         fi
+    else
+        packages+=(
+            'google-chrome-stable'
+        )
     fi
 
     sudo apt-get -qq remove -y --fix-missing "${packages[@]}"

